### PR TITLE
Woo-Connect Insights: Query and Normalize Orders Data

### PR DESCRIPTION
**Pull request by psealock**
### Query and Normalize Orders Data
In order render the main chart, the data will need to be queried and normalized.

screen shot 2017-05-22 at 1 12 04 pm

### Comments
Normalization is the exact same for `statsOrders` as it is for `statsVisits`. ~Until I figure out a better way to share that logic, I've hacked a solution.~ I pulled the logic into a function `parseChartData` so each normalizer can have access

~I've also printed out some data for now and will remove on a final clean up.~ Removed

### Also
I'm renaming `period` (days, weeks, months, years) to `unit` so that it matches the nomenclature of the API and backend.

Fixes https://github.com/Automattic/wp-calypso/issues/12984

### CODE REVIEW

File: client/state/stats/lists/utils.js [hunk 1]
```diff
@@ -620,51 +675,11 @@ export const normalizers = {
...
-					dataRecord.labelYear = localizedDate.format( 'YYYY' );
-				}
-			}
+		return parseChartData( payload, [ 'visits', 'likes', 'visitors', 'comments', 'posts' ] );
+	},
 
-			return dataRecord;
-		} );
+	statsOrders( payload ) {
+		return parseChartData( payload );
```
Comment (greenafrican) [inline client/state/stats/lists/utils.js]: This returns `null` however `return parseChartData( payload.orders );` works as the `data` object needed here (https://github.com/Automattic/wp-calypso/pull/14313/files#diff-6da59aec4d5693709df18761a243d4beR117) is in `payload.orders.data`.

**Comment by psealock** [comment]
@westi the response for `statsOrders` wraps the data in an object `{ orders: ... }`. In the Calypso context, it is extraneous. Can we remove it from the response?

**Comment by westi** [comment]
> @westi the response for statsOrders wraps the data in an object { orders: ... }. In the Calypso context, it is extraneous. Can we remove it from the response?

Removed the extra level of hierarchy for you

File: client/state/stats/lists/utils.js (hunk 1 above)
Comment (psealock) [inline client/state/stats/lists/utils.js]: Fixed in the API, the `payload` argument works as is.

File: client/extensions/woocommerce/app/stats/index.js [hunk 2]
```diff
@@ -2,19 +2,40 @@
  * External dependencies
  */
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
```
Comment (allendav) [inline client/extensions/woocommerce/app/stats/index.js]: Minor - we have been trying to keep the dependency import statements in alphabetical order elsewhere.

File: client/state/stats/lists/utils.js [hunk 3]
```diff
@@ -104,6 +104,61 @@ export function getSerializedStatsQuery( query = {} ) {
...
+	return payload.data.map( function( record ) {
+		// Initialize data
+		const dataRecord = nullAttributes.reduce( ( memo, attribute ) => {
+			memo[ attribute ] = null;
+			return memo;
+		}, {} );
+
+		// Fill Field Values
+		record.forEach( function( value, i ) {
+			// Remove W from weeks
```
Comment (allendav) [inline client/state/stats/lists/utils.js]: I'm curious - why is this needed only for weeks?

Comment (psealock) [inline client/state/stats/lists/utils.js]: I moved this code and did not want to touch it. A clue as to its use is this test:

https://github.com/Automattic/wp-calypso/blob/master/client/state/stats/lists/test/utils.js#L1525-L1532

File: client/extensions/woocommerce/app/stats/index.js [hunk 4]
```diff
@@ -2,19 +2,40 @@
...
 /**
  * Internal dependencies
  */
 import Main from 'components/main';
 import StatsNavigation from './stats-navigation';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import StatsChart from './stats-chart';
 
-export default class Stats extends Component {
+class Stats extends Component {
```
Comment (greenafrican) [inline client/extensions/woocommerce/app/stats/index.js]: Not critical but we may want to rename this at some point. `StatsStore` perhaps (similar to `StatsSite` [here](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/stats/site.jsx#L30))?

**Comment by greenafrican** [comment]
Works as expected. 

🚢

File: client/extensions/woocommerce/app/stats/index.js (hunk 4 above)
Comment (psealock) [inline client/extensions/woocommerce/app/stats/index.js]: This is a good point. Adding to this, due to eslint's jsx classname rules, there may be some conflicts in CSS namespaces with my current naming structure `stats-`. I hope to take a better look at naming in a separate PR.
